### PR TITLE
Collapsible ToC for increased accessibility

### PR DESCRIPTION
**Pull request by rubek-joshi**
The amount of content here has gotten quite long and is expected to keep growing. Hence, I think it's time to add the sub-sections to the table of contents. 

I know there is a `CTRl + F` thingy but I think sometimes you just want to look up the subtopics list and navigate to it for a quick refresher. Just me?

### CODE REVIEW

**Review by elektronik2k5 — APPROVED** [review]

**Comment by goldbergyoni** [comment]
@rubek-joshi Just got back from a long vacation, really love the idea!

@elektronik2k5 @josh-hemphill @rluvaton What do we think about this?

**Comment by elektronik2k5** [comment]
I'm definitely in favor!

File: README.md [hunk 1]
```diff
@@ -51,14 +50,171 @@ Read in a different language: [![CN](./assets/flags/CN.png)**CN**](./README.chin
...
+<details>
+  <summary>
+    <a href="#1-project-structure-practices">1. Project Structure Practices (5)</a>
+  </summary>
+
+&emsp;&emsp;<a href="#-11-structure-your-solution-by-components">1.1 Structure your solution by components</a></br>
+&emsp;&emsp;<a href="#-12-layer-your-components-keep-the-web-layer-within-its-boundaries">1.2 Layer your components, keep the web layer within its boundaries</a></br>
+&emsp;&emsp;<a href="#-13-wrap-common-utilities-as-npm-packages">1.3 Wrap common utilities as npm packages</a></br>
+&emsp;&emsp;<a href="#-14-separate-express-app-and-server">1.4 Separate Express 'app' and 'server'</a></br>
+&emsp;&emsp;<a href="#-15-use-environment-aware-secure-and-hierarchical-config">1.5 Use environment aware, secure and hierarchical config</a></br>
```
Comment (elektronik2k5) [inline README.md:62]: Can we have the old markup here back, please? We're using MD and not HTML. I've successfully used details/summary with MD inside on GitHub in the past.

**Review by elektronik2k5 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (rluvaton) [reply README.md:62]: Look [here](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab) for example

**Comment by rluvaton** [comment]
> @rluvaton What do we think about this?

Looks great!

**Comment by goldbergyoni** [comment]
@rubek-joshi @rluvaton @elektronik2k5 

Since we're doing this big change, how about we solve one more problem: 

1. It's hard to focus on the bullets that are even more important
2. When re-visiting, hard to tell what's new and it's tedious to re-read all over again

Solution - We add tags to some items, included two different formats  👇

image

File: README.md (hunk 1 above)
Comment (rubek-joshi) [reply README.md]: I've managed to revert back the sub headers to the old markup but I couldn't get it working for the main header (i.e. inside the `<summary/>` tag).

**Comment by rubek-joshi** [comment]
> @rubek-joshi @rluvaton @elektronik2k5
> 
> Since we're doing this big change, how about we solve two more problem:
> 
> 1. It's hard to prioritize and focus on bullets that are more important
> 2. When re-visiting this guide, it's hard to tell what's new and it's tedious to re-read all over again
> 
> Solution - We add tags to some items. I've included two different formats 👇
> 
> Thoughts?
> 
> image

This is a great idea but I wouldn't know what tags to put where unless someone provides me info.

**Comment by goldbergyoni** [comment]
Me and others can do the tagging, question is wether everyone love it

**Comment by rubek-joshi** [comment]
I don't see why anyone wouldn't like it. Do it!

**Comment by goldbergyoni** [comment]
@rubek-joshi Let's merge your amazing change once we switch to markup? If have additional 5min, just ensure all work, maybe sample some links and click

**Comment by rubek-joshi** [comment]
@goldbergyoni Just tested and all the links are working correctly, I think we're good to go now

**Comment by goldbergyoni** [comment]
@rubek-joshi Awesome. Excited to see this in action

@elektronik2k5 Collapsible units are now implemented with markdown, can I approve?

**Review by elektronik2k5 — APPROVED** [review]

**Pull request merged**